MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpressHci.c

```diff
@@ -732,15 +1039,23 @@ NvmeControllerInit (
   IN NVME_CONTROLLER_PRIVATE_DATA  *Private
   )
 {
+  // MU_CHANGE [BEGIN] - Allocate IO Queue Buffer
   EFI_STATUS           Status;
   EFI_PCI_IO_PROTOCOL  *PciIo;
   UINT64               Supports;
-  NVME_AQA             Aqa;
-  NVME_ASQ             Asq;
-  NVME_ACQ             Acq;
-  UINT16               VidDid[2]; // MU_CHANGE - Improve NVMe controller init robustness
-  UINT8                Sn[21];
-  UINT8                Mn[41];
+  UINT16               VidDid[2];  // MU_CHANGE - Improve NVMe controller init robustness
+  // NVME_ASQ             Asq;
+  // NVME_ACQ             Acq;
+  UINT8                 Sn[21];
+  UINT8                 Mn[41];

```

Comment:
Where do these magic numbers come from?

Reply:
Serial Number & Model number in this code are copied from the NVME device and are assumed to be 40 bytes in length with the 41st byte reserved for null terminator (index 40)

Reply:
https://nvmexpress.org/specifications/

Nvme Express base Specification section  4.5 Identifier Format and Layout (Informative)
The strings provided by the identify response are non -null terminated.

Reply:
From Nvm Express Base Spec 4.3.2
> The Serial Number (SN, bytes 23:04) and Model Number (MN, bytes 63:24) are defined in the Identify 
Controller data structure. The values are ASCII strings assigned by the vendor.

Reply:
Line 1525 has a comment where we null-terminate the result